docs/en/operations/server-configuration-parameters/_server_settings_outside_source.md

```diff
@@ -2448,6 +2448,60 @@ The same applies to GitLab, even though it has a leading dot. Both `gitlab.com`
 </proxy>
 ```
 
+## vault (#vault)
+
+Configures OpenBao or HashiCorp Vault for secure secrets retrieval in configuration. Currently only token authentication is supported.
+
+The following settings can be configured by sub-tags:
+
+| Sub-tags             | Definition                                                                                                                                        |
+|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
+| `url`                | Scheme, site, port of Vault server without path.                                                           |
+| `token`                | Token used for token authentication with Vault                                                           |
+
+Each section in ClickHouse configuration or users configuration may have the following attributes:
+
+| Attributes             | Definition                                                                                                                                        |
+|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
+| `from_vault`                 | Name of secret.                                                           |
+| `vault_key`                | Name of secret's key                                                           |
+
+**Example**
+
+ClickHouse configuration:
+
+```xml
+<clickhouse>
+    <vault>
+      <url>http://openbao:1337</url>
+      <token>foobar</token>
+    </vault>
+</clickhouse>
+```
+
+Users configuration:
+
+```xml
+<clickhouse>
+    <users>
+        <default>
+            <password from_vault="ch_secret" vault_key="password"/>
```

Comment:
What does 'ch_secret' (Name of secret) exactly mean?

Reply:
Done